src/components/AutoUpdateTime.js

```diff
@@ -22,85 +27,70 @@ const propTypes = {
     ...withLocalizePropTypes,
 };
 
-class AutoUpdateTime extends PureComponent {
-    constructor(props) {
-        super(props);
-        this.getCurrentUserLocalTime = this.getCurrentUserLocalTime.bind(this);
-        this.updateCurrentTime = this.updateCurrentTime.bind(this);
-        this.getTimezoneName = this.getTimezoneName.bind(this);
-        this.state = {
-            currentUserLocalTime: this.getCurrentUserLocalTime(),
-        };
-    }
-
-    componentDidMount() {
-        this.updateCurrentTime();
-    }
-
-    componentDidUpdate() {
-        // Make sure the interval is up to date every time the component updates
-        this.updateCurrentTime();
-    }
-
-    componentWillUnmount() {
-        clearTimeout(this.timer);
-    }
-
+function AutoUpdateTime(props) {
     /**
      * @returns {moment} Returns the locale moment object
      */
-    getCurrentUserLocalTime() {
-        return DateUtils.getLocalMomentFromDatetime(
-            this.props.preferredLocale,
+    const getCurrentUserLocalTime = useCallback(() => (
+        DateUtils.getLocalMomentFromDatetime(
+            props.preferredLocale,
             null,
-            this.props.timezone.selected,
-        );
-    }
+            props.timezone.selected,
+        )
+    ), [props.preferredLocale, props.timezone.selected]);
+
+    const [currentUserLocalTime, setCurrentUserLocalTime] = useState(getCurrentUserLocalTime());
```

Comment:
A weird thing about `useState()` is that it will keep calling `getCurrentUserLocalTime()` on every render even though it only uses the initial value once. I don't really get why, but to stop that we should do:

```suggestion
    const [currentUserLocalTime, setCurrentUserLocalTime] = useState(getCurrentUserLocalTime);
```

and now React will only call this function once.

https://react.dev/reference/react/useState#avoiding-recreating-the-initial-state

Reply:
Good to know! Updated.